src/components/dialogs/field-hook.js

```diff
@@ -61,11 +61,13 @@ export const useTextValue = ({
     ...formProps
 }) => {
     const [value, setValue] = useState(defaultValue);
+    const [touched, setTouched] = useState(false);
```

Comment:
Maybe you could name this hook hasChanged, setHasChanged to be homogeneous with gridstudy-app